.github/workflows/cypress-test-multiauth-e2e.yml

```diff
@@ -82,14 +83,30 @@ jobs:
           opensearch_security.multitenancy.tenants.preferred: ["Private", "Global"]
           opensearch_security.readonly_mode.roles: ["kibana_read_only"]
           opensearch_security.cookie.secure: false
-          server.xsrf.allowlist: ["/_plugins/_security/api/authtoken", "/_opendistro/_security/api/authtoken", "/_opendistro/_security/saml/acs", "/_opendistro/_security/saml/acs/idpinitiated", "/_opendistro/_security/saml/logout"]
+          server.xsrf.allowlist: ["/_opendistro/_security/saml/acs", "/_opendistro/_security/saml/acs/idpinitiated", "/_opendistro/_security/saml/logout"]
           opensearch_security.auth.type: ["basicauth","saml"]
           opensearch_security.auth.multiple_auth_enabled: true
           opensearch_security.auth.anonymous_auth_enabled: false
           home.disableWelcomeScreen: true
           EOT
 
+      - name: Run OSD with basePath
+        if: ${{ matrix.basePath != '' }}
+        run: |
+          echo "server.basePath: \"${{ matrix.basePath }}\"" >> opensearch_dashboards_multiauth.yml
+          echo "server.rewriteBasePath: true" >> opensearch_dashboards_multiauth.yml
```

Comment:
Is this necessary for testing? https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/config/opensearch_dashboards.yml#L17 I am not familiar with this setting but I dont think this action would be testing the proxy flow?

Reply:
Good question. This setting is used to tell OSD if it should rewrite requests to include the basePath if it does not already (i.e. localhost:5601 gets redirected to localhost:5601/{basePath}). If set to false, it assumes OSD is behind a reverse proxy and that the proxy will rewrite the requests. If set to false and a user navigates to root path instead of the basePath, they won't get redirected into OSD.

Reply:
I have generally set it to true when working with base paths.